src/drive/web/modules/navigation/duck/actions.jsx

```diff
@@ -502,3 +504,14 @@ export const openFileWith = (id, filename) => {
     }
   }
 }
+
+export const deriveEncryptionKey = passphrase => {
+  return async (dispatch, _, { client }) => {
+    const salt = client.getStackClient().uri
+    const key = await deriveKey(passphrase, salt)
+    return dispatch({
+      type: DERIVE_ENCRYPTION_KEY,
```

Comment:
I don't understand/know much about Redux behavior but ... Does `DERIVE_ENCRYPTION_KEY` represent an *action* or an *object* ? Here is my point of view :
- dispatch is called on an object holding the an attribute "derive_encryption_key" and an **already deriveD** key
- so the dispatch holds a derived key
- so why not having `type: DERIVED_ENCRYPTION_KEY` (with 'D')
?

Reply:
It represents the type of the action. From redux doc: 
> Actions are payloads of information that send data from your application to your store.

And the `dispatch` method only accept an action as argument: https://redux.js.org/api/store#dispatch

We dispatch the action to the reducer, where the action of the type `DERIVE_ENCRYPTION_KEY` will be treated accordingly (save the derived key to the store). So the action *is* `DERIVE_ENCRYPTION_KEY`, where the payload is the result of `deriveKey`. 


This is the same logic as the others actions, see [downloadFile](https://github.com/Gara64/cozy-drive/blob/83b5096d9da6ff10e5f97bcbb6c56b2eedd0115f/src/drive/web/modules/navigation/duck/actions.jsx#L428-L440) for instance.